plugin/kubernetes/containerBase.go

```diff
@@ -0,0 +1,108 @@
+package kubernetes
+
+import (
+	"context"
+	"io"
+
+	"github.com/puppetlabs/wash/activity"
+	corev1 "k8s.io/api/core/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+// A general purpose container object that implements executing commands.
+// If the pod contains only a single container, the container object may be omitted.
+type containerBase struct {
+	client    *k8s.Clientset
+	config    *rest.Config
+	pod       *corev1.Pod
+	container *corev1.Container
+}
+
+// Create an executor to run a command. If you want synchronous results, call executor.Stream(). For
+// asynchronous results call the `async` method.
+func (c *containerBase) exec(ctx context.Context, cmd string, args []string, opts remotecommand.StreamOptions) (remotecommand.Executor, error) {
```

Comment:
`exec` => `newExecutor` ? Matches what's returned, and a bit less confusing b/c it doesn't conflate "exec a command" vs. "exec as in create a new executor".